core/include/scipp/core/element/event_operations.h

```diff
@@ -16,74 +16,93 @@
 
 namespace scipp::core::element::event {
 
-namespace map_in_place_detail {
+constexpr auto get = [](const auto &x, const scipp::index i) {
+  if constexpr (is_ValueAndVariance_v<std::decay_t<decltype(x)>>)
+    return ValueAndVariance{x.value[i], x.variance[i]};
```

Comment:
If ValueAndVariance had the subscript operator overload would collapse to a single line. Any reason why we could not have this?

Reply:
It would require specializing `ValueAndVariance` for `span` and other classes. Sounds like an interesting option. On the other hand, the need for this is for me still and indicator that we should handle these things differently. The `subspan_view` approach is working, but does not seem ideal (and in fact I feel I am often hitting its limits). Still hoping to come up with an overall cleaner approach.